packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -91,16 +105,45 @@ export default class PayPalCommercePaymentStrategy implements PaymentStrategy {
             throw new PaymentArgumentInvalidError(['payment']);
         }
 
+        const { paymentData } = payment;
+        const isVaultedFlow =
+            paymentData && isVaultedInstrument(paymentData) && isHostedInstrumentLike(paymentData);
+
         if (!this.orderId) {
-            throw new PaymentMethodInvalidError();
+            if (isVaultedFlow) {
+                this.orderId = await this.createOrder();
+            } else {
+                throw new PaymentMethodInvalidError();
+            }
+        }
+
+        let isTrustedVaultingFlow = true;
```

Comment:
Why `isTrustedVaultingFlow` has true as default value?

Reply:
Did it another way. Take a look at the updates